atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/query/forming/UnifiedCacheQueryFormer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.async.query.forming;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+
+public final class UnifiedCacheQueryFormer extends AbstractQueryFormer {
```

Comment:
Not sure the difference between the two impls here.

Reply:
This one has a single cache for all queries we can add in the future, the second one is implemented such that each query type has its own cache.